framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2311,6 +2311,17 @@ public boolean shouldSuppressWarnings(@Nullable TreePath path, String errKey) {
                     // elements, because they may not have an @AnnotatedFor.
                     return false;
                 }
+                Element packageElement = elt.getEnclosingElement();
```

Comment:
A class could have another class as enclosing element. Or even a method, for a local class or an anonymous class.
`TreePathUtil.enclosingDeclarationPath` is only used here. I think it would be nicest to expand its specification to also return the package, if there is one. Then you can just have another branch for packages.

Reply:
@wmdietl I think the problem is not only about `TreePathUtil.enclosingDeclarationPath` but also for `declPath.getParentPath()` does not contain the package in the parent path. That's why I am adding the checks here. 

If class or method  is enclosing element for current class, `TreePathUtil.enclosingDeclarationPath` will iterate and get the new `declPath`. So it will go through the all the if else check again util we get to the outermost class and check the package-info.

Another possibility is we don't iterate over the treepath and recursively check whether the enclosing element. Maybe that would be nicer?

Reply:
Thanks for your code suggestion. I think right now this should be the solution if we want to keep iterate over treepath's parent path. I am wondering is there any difference between we iterate over the treepath's parent and element's enclosing element? If there is no difference, we can directly get the leaf and get the element, and checking the enclosing element then.

Reply:
I'm not sure an enclosing tree is always the same as the enclosing element, but you could investigate that. As this is working now, let's merge this.